### PR TITLE
Fix NPE when importing OAS3 files

### PR DESCRIPTION
**Pull request by RakhithaRR**
This is the public fix for https://github.com/wso2/product-apim/issues/9063.

### CODE REVIEW

**Review by tgtshanika — DISMISSED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java [hunk 1]
```diff
@@ -1458,6 +1460,14 @@ private OpenAPI injectOtherScopesToDefaultScheme(OpenAPI openAPI) throws APIMana
             if (defaultScheme == null) {
                 SecurityScheme newDefault = new SecurityScheme();
                 newDefault.setType(SecurityScheme.Type.OAUTH2);
+                OAuthFlows newDefaultFlows = new OAuthFlows();
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java:1463]: Shall we add a comment on what we are doing here. ie: Populating the default security scheme.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java [hunk 2]
```diff
@@ -1458,6 +1460,14 @@ private OpenAPI injectOtherScopesToDefaultScheme(OpenAPI openAPI) throws APIMana
...
+                OAuthFlow newDefaultFlow = new OAuthFlow();
+                newDefaultFlow.setAuthorizationUrl("https://test.com");
+                Scopes newDefaultScopes = new Scopes();
+                newDefaultFlow.setScopes(newDefaultScopes);
+                newDefaultFlows.setImplicit(newDefaultFlow);
+                newDefault.setFlows(newDefaultFlows);
+
                 securitySchemes.put(OPENAPI_SECURITY_SCHEMA_KEY, newDefault);
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java:1471]: Check the same flow with Swagger 2 API definition and fix it in OAS2Parser, if the same issue is there.

Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java:1471]: Default security is populated in v2 parser. So, this issue is not there.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java (hunk 1 above)
Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java:1464]: fixed with 0d3ff9c

**Review by tgtshanika — APPROVED** [review]

**Pull request merged**